okhttp/src/jvmMain/kotlin/okhttp3/internal/connection/ExchangeFinder.kt

```diff
@@ -259,6 +225,7 @@ class ExchangeFinder(
     }
 
     // List available IP addresses for the current proxy. This may block in Dns.lookup().
+    if (!newRouteSelector.hasNext()) throw IOException("exhausted all routes")
```

Comment:
I'm not 100% convinced this is strictly necessary, but it’s better to fail this way than with a `NoSuchElementException`. (I hope to delete this later.)